utilities/decrypt_data/decrypt_multi.py

```
@@ -0,0 +1,237 @@
+    key_cols = table_config["key_columns"]
+    data_cols = table_config["columns"]
+    table = table_config["table"]
+
+    # Select both key columns + any non-null data columns
+    select_cols_sql = ", ".join(key_cols + data_cols)
+    where_data_not_null = " OR ".join(f"{c} IS NOT NULL" for c in data_cols)
+    select_sql = f"SELECT {select_cols_sql} FROM {table} WHERE {where_data_not_null}"
+
+    try:
+        with psycopg2.connect(CONFIG["db"]["dsn"]) as conn:
+            with conn.cursor(cursor_factory=DictCursor) as cur:
```

Ruff (0.12.2)

Use a single `with` statement with multiple contexts instead of nested `with` statements

(SIM117)